app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -74,27 +74,33 @@ class UndertakingController @Inject() (
     store.get[UndertakingJourney].flatMap {
       case Some(journey) =>
         val form = journey.name.value.fold(undertakingNameForm)(name => undertakingNameForm.fill(FormValues(name)))
-        Ok(undertakingNamePage(form)).toFuture
+        Ok(undertakingNamePage(form, journey.previous)).toFuture
       case None => // initialise the empty Journey model
-        store.put(UndertakingJourney()).map { _ =>
-          Ok(undertakingNamePage(undertakingNameForm))
+        val journey = UndertakingJourney()
+        store.put(journey).map { _ =>
+          Ok(undertakingNamePage(undertakingNameForm, UndertakingJourney().previous))
```

Comment:
May be a bit of a dumb question - But have you tested this? I'd thought it would have thrown an exception as there is no previous on a fresh journey?

Reply:
yes, I have tested this. If you look closely I have changed the previous logic in Undertaking Journey a little bit. I have created a map
```
Map(
    routes.UndertakingController.getUndertakingName().url -> routes.EligibilityController.getCreateUndertaking().url
  )
```
  which hold the previous for undertaking name page and instead of `super.previous` , here is 
  `previousMap.get(r.uri).getOrElse(super.previous) `now.
   As of now it's just one value in map but In future if there is a need to find the previous of a page which is not a part of the journey we can insert that value in map .